lib/src/rules/prefer_int_value_is_even_check.dart

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) 2020, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+
+import '../analyzer.dart';
+import '../util/boolean_expression_utilities.dart';
+
+const _desc =
+    r'Prefer intValue.isOdd/isEven instead of checking the result of % 2.';
+
+const _details = r'''
+
+**PREFER** the use of intValue.isOdd/isEven to check for evenness.
+
+**BAD:**
+```
+bool isEven = 1 % 2==0;
+bool isOdd = 13 % 2 ==1;
+```
+
+**GOOD:**
+```
+bool isEven =1.isEven;
+bool isOdd = 13.isOdd;
+```
+
+''';
+
+class PreferIntValueIsEvenCheck extends LintRule implements NodeLintRule {
+  PreferIntValueIsEvenCheck()
+      : super(
+            name: 'prefer_int_value_is_even_check',
```

Comment:
You had no way of knowing, but we've decided that it was a mistake to include prefixes like "prefer" in lint names. The reason is that the style guide can change over time, and things that were once discouraged might later be forbidden. Consider something like "use_is_even_rather_than_modulo" (alternative suggestions are welcome).

Reply:
> use_is_even_rather_than_modulo

Great name.  I was grasping for one and this tops it.  👍

Reply:
thanks for the name suggestion. Done